### PR TITLE
FEAT: `float_power` ufunc implementation

### PR DESCRIPTION
**Pull request by SwayamInSync**
## Copilot Summary
This pull request adds full support for the `float_power` ufunc to the quad-precision dtype implementation, ensuring its behavior matches NumPy's expectations for floating-point types. The changes include the implementation of the `float_power` operation, updates to documentation, and comprehensive tests to verify correctness and edge cases.

**Quad-precision ufunc implementation:**

* Added support for the `float_power` ufunc in `init_quad_binary_ops`, using the same implementation as `power`, since quad-precision is already a floating-point type.

**Documentation and release tracking:**

* Updated `release_tracker.md` to mark `float_power` as implemented and tested, reflecting its new support status.

**Testing and validation:**

* Added extensive parameterized tests for `float_power` in `test_quaddtype.py`, covering a wide range of cases including integer and fractional exponents, negative bases, zero and infinity handling, NaN propagation, integer promotion, and array operations. These tests ensure the quad-precision implementation matches NumPy's behavior for all relevant scenarios.

### CODE REVIEW

File: quaddtype/numpy_quaddtype/src/umath/binary_ops.cpp [hunk 1]
```diff
@@ -224,6 +224,11 @@ init_quad_binary_ops(PyObject *numpy)
     if (create_quad_binary_ufunc<quad_pow, ld_pow>(numpy, "power") < 0) {
         return -1;
     }
+    // float_power uses the same implementation as power for floating-point types
+    // The only difference is that float_power promotes integer inputs to float (quaddtype already float)
```
Comment (juntyr) [inline quaddtype/numpy_quaddtype/src/umath/binary_ops.cpp]:
```suggestion
    // The only difference is that float_power promotes integer inputs to float (quaddtype is already float)
```

**Comment by SwayamInSync** [comment]
Thanks @juntyr , merging this
